code/modules/station_goals/secondary/variety_reagent.dm

```diff
@@ -45,27 +45,22 @@
 	return copy
 
 /datum/secondary_goal_progress/variety_reagent/update(atom/movable/AM, datum/economy/cargo_shuttle_manifest/manifest = null)
-	// Not a reagent container? Ignore.
-	if(!istype(AM, /obj/item/reagent_containers))
-		return
-
 	// Not in a matching personal crate? Ignore.
 	if(!check_personal_crate(AM))
 		return
 
-	var/obj/item/reagent_containers/container = AM
 	// No reagents? Ignore.
-	if(!container.reagents.reagent_list)
+	if(!AM.reagents?.reagent_list)
 		return
```

Comment:
Shouldn't this check the length of the list?

Reply:
Nah, the !reagent check a few lines down has that covered.